### PR TITLE
Allow passing versioned media types to 7.x server

### PR DESCRIPTION
**Pull request by pgomulka**
7.x client can pass media type with a version which will return a 7.x
version of the api in ES 8.
In ES server 7 this media type shoulld be accepted but it serve the same
version of the API (7x)
relates #51816

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/2

File: libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentType.java [hunk 1]
```diff
@@ -137,6 +139,20 @@ public static XContentType fromMediaTypeOrFormat(String mediaType) {
...
+    /**
+     * Clients compatible with ES 7.x might start sending media types with versioned media type
+     * in a form of application/vnd.elasticsearch+json;compatible-with=7.
+     * This has to be removed in order to be used in 7.x server.
+     * The same client connecting using that media type will be able to communicate with ES 8 thanks to compatible API.
+     * @param mediaType - a media type used on Content-Type header, might contain versioned media type.
+     *
+     * @return a media type string without
+     */
+    private static String removeVersionInMediaType(String mediaType) {
```
Comment (jakelandis) [inline libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentType.java:151]: I would suggest to wrap this in if(mediaType.contains("vnd.elasticsearch"){ ...} else return mediaType.  It is a micro-optimization, but will probably save a few CPU cycles. Also I think you can collapse the two replaceAll into a single line.

**Review by jakelandis — APPROVED** [review]
LGTM, 1 minor optional comment

**Review by jaymode — APPROVED** [review]
LGTM

**Pull request merged**